src/components/ArticleList.jsx

```diff
@@ -21,19 +23,36 @@ function ArticleList() {
             break;
     }
     
-    const sortArticles = useCallback((articles, sortQuery) => {
-        const sortedArticles = articles.sort((a, b) => {
-          if (sortQuery === "created_at") {
-            return new Date(b[sortQuery]) - new Date(a[sortQuery]);
-          }
-          if (Number(a[sortQuery]) < Number(b[sortQuery])) {
-            return 1;
-          }
-          if (Number(a[sortQuery]) > Number(b[sortQuery])) {
-            return -1;
-          }
-          return 0;
-        });
+    const sortArticles = useCallback((articles, sortQuery, sortAscending) => {
+        setSortKey(sortQuery)
+        let sortedArticles
+        if (!sortAscending) {
+            sortedArticles = articles.sort((a, b) => {
+                if (sortQuery === "created_at") {
+                  return new Date(b[sortQuery]) - new Date(a[sortQuery]);
+                }
+                if (Number(a[sortQuery]) < Number(b[sortQuery])) {
+                  return 1;
+                }
+                if (Number(a[sortQuery]) > Number(b[sortQuery])) {
+                  return -1;
+                }
+                return 0;
+              });
+        } else {
+            sortedArticles = articles.sort((a, b) => {
+                if (sortQuery === "created_at") {
+                  return new Date(a[sortQuery]) - new Date(b[sortQuery]);
+                }
+                if (Number(a[sortQuery]) < Number(b[sortQuery])) {
+                  return -1;
+                }
+                if (Number(a[sortQuery]) > Number(b[sortQuery])) {
+                  return 1;
+                }
+                return 0;
+              });
+        }
```

Comment:
Because you are checking for created_at in both cases here I think it's only working for Date. If you are setting the sortQuery correctly, I would use that value in your sort and I feel it could be done without the if/else if you are then just using whichever value it is. It gets tricky between the different values because they may well be different data types which might trip you up. I would also advise maybe extracting this out into it's own utils file as it's taking up quite a bit of space in this component! 

This is also trickier to do without the back end having the capability, so if you ever decide to have a look at how to implement that you wouldn't need this! Good effort though.